components/raftstore/src/store/peer.rs

```diff
@@ -1884,12 +1886,20 @@ where
         let msg_type = m.get_msg_type();
         if msg_type == MessageType::MsgReadIndex {
             fail_point!("on_step_read_index_msg");
+            let mut start_ts: u64 = 0;
+            if !m.get_entries().is_empty() && !m.get_entries()[0].get_data().is_empty() {
+                let mut rctx = ReadIndexContext::parse(m.get_entries()[0].get_data()).unwrap();
```

Comment:
I think for read index request, the entries always is empty.

Reply:
no, it has the timestamp and key range. we read this in on_step_read_index() also

Reply:
i have made more code changes which are not in this PR. You can take a look at the latest change here . https://github.com/mittalrishabh/tikv/blob/7fb5feabe9a659c079262b856cc990da364bfd83/components/raftstore/src/store/peer.rs#L1840

Reply:
Can you provide which branch? I'd like to review the whole change history.

Reply:
https://github.com/mittalrishabh/tikv/commits/release-7.5.1